scripts/find.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+# Copyright 2020 The Rust Project Developers
+#
+# Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+# http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+# http://opensource.org/licenses/MIT>, at your option. This file may not be
+# copied, modified, or distributed except according to those terms.
+
+NEEDLE="E0642"
+
+! cargo build 2>&1 | grep "$NEEDLE"
```

Comment:
I would suggest to change this to use [parameter expansion](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) and let the variable set from the outer world, so this script would be more generic and you can do stuff like:

`env NEEDLE="E0600" scripts/find.sh`

instead of manually changing the script.

So something like:

```
! cargo build 2>&1 | grep "${NEEDLE:-E0642}"
```

should do the trick

Reply:
Great idea! Thoughts about using a regex for any compiler error code as the default? Is that a common enough task or is there a better generally useful default fallback when env var is not set?

Reply:
Holding on edits until https://github.com/rust-lang/cargo-bisect-rustc/pull/47#issuecomment-584904257 is settled.